.github/workflows/validate-mkdocs.yml

```diff
@@ -0,0 +1,29 @@
+name: Valdate static MkDocs pages
+on: pull_request
+
+jobs:
+  validate-mkdocs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
What does fetching with a depth of 0 do? It seems nonsensical according to the git-fetch manpage:
> −−depth=&lt;depth&gt;
Limit fetching to the specified number of commits from the tip of each remote branch history. If fetching to a shallow repository created by git clone with −−depth=<depth> option (see git-clone(1)), deepen or shorten the history to the specified number of commits. Tags for the deepened commits are not fetched.

Reply:
Fetches all tags and branches: https://github.com/marketplace/actions/checkout#fetch-all-history-for-all-tags-and-branches

It technically only needs the most recent commits in `master` (for building the docs) and `gh-pages` (for deploying the docs) but I don't think there's an easy way to do that

Reply:
_0_ fetches everything??? Wow, GitHub really dropped the ball on the semantics here.